pkg/process/events/store_test.go

```diff
@@ -0,0 +1,195 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package events
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-go/v5/statsd"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/process/events/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// pushSync is an auxiliary function to push events to the RingStore synchronously
+func pushSync(t *testing.T, r *RingStore, e *model.ProcessEvent) {
+	done := make(chan bool)
+	r.Push(e, done)
+	ok := <-done
+	require.True(t, ok)
+}
+
+func TestRingStore(t *testing.T) {
```

Comment:
It appears this test is doing too much for a single test, what are your thoughts about splitting it? state can be initialized directly by adding nodes in the struct.